checker/src/org/checkerframework/checker/lock/LockTreeAnnotator.java

```diff
@@ -0,0 +1,53 @@
+package org.checkerframework.checker.lock;
+
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.util.Elements;
+
+import org.checkerframework.checker.lock.qual.GuardedBy;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.treeannotator.TreeAnnotator;
+import org.checkerframework.javacutil.AnnotationUtils;
+
+import com.sun.source.tree.BinaryTree;
+import com.sun.source.tree.Tree.Kind;
+
+public class LockTreeAnnotator extends TreeAnnotator {
+
+    /** Annotation constants */
+    protected final AnnotationMirror GUARDEDBY;
+
+    public LockTreeAnnotator(AnnotatedTypeFactory atypeFactory) {
+        super(atypeFactory);
+
+        Elements elements = atypeFactory.getElementUtils();
+        GUARDEDBY = AnnotationUtils.fromClass(elements, GuardedBy.class);
+    }
+
+    @Override
+    public Void visitBinary(BinaryTree node, AnnotatedTypeMirror type) {
+        if (isBinaryComparisonOperator(node.getKind())) {
```

Comment:
Why is this just for binaryComparisons? In other words, when should the type of a BinaryTree not be `@GuardedBy({})`?  (At the very least String concatenations, should be `@GuardedBy({})`.)